eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -463,19 +404,26 @@ public synchronized BackupRegistry get() {
     }
 
     private void scheduleServerEndpointTask(EurekaTransport eurekaTransport,
-                                            DiscoveryClientOptionalArgs args) {
+                                            AbstractDiscoveryClientOptionalArgs args) {
 
-        Collection<ClientFilter> additionalFilters = args == null
-                ? Collections.<ClientFilter>emptyList()
+            
+        Collection<?> additionalFilters = args == null
+                ? Collections.emptyList()
                 : args.additionalFilters;
 
         EurekaJerseyClient providedJerseyClient = args == null
                 ? null
                 : args.eurekaJerseyClient;
-
+        
+        TransportClientFactories transportClientFactories = TransportClientFactories.INSTANCE;
```

Comment:
Can we try another way to do this here instead of having a default static instance reference of a child class in the TransportClientFactories interface? One suggestion could be:
1. define the interface for TransportClientFactories as is, but with no default static instance
2. have a TransportClientFactoriesProvider, e.g. 
```java
public class TransportClientFactoriesProvider implements Provider<TransportClientFactories<?>> {

    private volatile TransportClientFactories<?> transportClientFactories;

    public TransportClientFactoriesProvider(TransportClientFactories<?> transportClientFactories) {
        this.transportClientFactories = transportClientFactories;
    }

    @Override
    public synchronized TransportClientFactories get() {
        if (transportClientFactories == null) {
            transportClientFactories = new Jersey1TransportClientFactories();
        }

        return transportClientFactories;
    }
}
```

3. in DiscoveryClient, initialize this provider with args.getTransportClientFactories(), then just get the actual impl out here in the same way.